### ISSUE TITLE
No suggestions in spellcheck when using Solr Net Cloud Unity component 

### ISSUE THREAD

**Issue by Fatalityap** [SolrNet/SolrNet#397]
Hi I have following code. And it returns zero suggestions while in Fiddler (when I have captured Solr Net call) I see **one suggestion**. Can you help me? 

```
        public async Task<List<NumberPlate>> FindNumberPlate(string keyword)
        {
            var provider = new SolrCloudStateProvider(ZookeeperConnection);
            await provider.InitAsync();
            var state = provider.GetCloudState();
            var colcount = state.Collections.Count;

            using (var container = new UnityContainer())
            {
                var cont = new SolrNetContainerConfiguration().ConfigureContainer(provider, container);
                var obj = cont.Resolve<ISolrOperations<NumberPlate>>("numberplate");
                //return await obj.QueryAsync(new SolrQueryByField("Organization", keyword));

                

                var queryResult = await obj.QueryAsync(new SolrQuery(keyword), new QueryOptions {
                    SpellCheck = new SpellCheckingParameters { Collate = true },
                    ExtraParams = new Dictionary<string, string>
                    {
                        {"defType","dismax"},
                        {"qf", "Number^3 Organization^2 FirstName^1.5 LastName^1.3 CountryOfRegistration^1.2"},
                        {"spellcheck", "on"},
                        {"spellcheck.q", keyword}
                    }
                });

               var spellSuggest = queryResult.SpellChecking.First().Suggestions.Count(); //Zero suggestions while in Fiddler there is one! 

                foreach(var result in queryResult)
                {
                   foreach(var sc in queryResult.SpellChecking)
                   {
                      result.SpellCheckRes = sc.Suggestions.ToList();
                   }
                }

                return queryResult;
            }
        }
```

Here is my **request in Fiddler**  (when I have captured Solr Net call)

`GET /solr/numberplate/select?q=Ponomarenko1&rows=100000000&defType=dismax&qf=Number%5e3+Organization%5e2+FirstName%5e1.5+LastName%5e1.3+CountryOfRegistration%5e1.2&spellcheck=on&spellcheck.q=Ponomarenko1&spellcheck=true&spellcheck.collate=true&version=2.2&wt=xml HTTP/1.1`

Here is my **response in Fiddler**  (when I have captured Solr Net call)

```
<?xml version="1.0" encoding="UTF-8"?>
<response>

<lst name="responseHeader">
  <bool name="zkConnected">true</bool>
  <int name="status">0</int>
  <int name="QTime">72</int>
  <lst name="params">
    <str name="q">Ponomarenko1</str>
    <str name="defType">dismax</str>
    <str name="spellcheck.q">Ponomarenko1</str>
    <str name="qf">Number^3 Organization^2 FirstName^1.5 LastName^1.3 CountryOfRegistration^1.2</str>
    <arr name="spellcheck">
      <str>on</str>
      <str>true</str>
    </arr>
    <str name="rows">100000000</str>
    <str name="version">2.2</str>
    <str name="wt">xml</str>
    <str name="spellcheck.collate">true</str>
  </lst>
</lst>
<result name="response" numFound="0" start="0" maxScore="0.0">
</result>
<lst name="spellcheck">
  <lst name="suggestions">
    <lst name="Ponomarenko1">
      <int name="numFound">1</int>
      <int name="startOffset">0</int>
      <int name="endOffset">12</int>
      <int name="origFreq">0</int>
      <arr name="suggestion">
        <lst>
          <str name="word">Ponomarenko</str>
          <int name="freq">11</int>
        </lst>
      </arr>
    </lst>
  </lst>
  <bool name="correctlySpelled">false</bool>
  <lst name="collations">
    <lst name="collation">
      <str name="collationQuery">Ponomarenko</str>
      <long name="hits">11</long>
      <lst name="misspellingsAndCorrections">
        <str name="Ponomarenko1">Ponomarenko</str>
      </lst>
    </lst>
  </lst>
</lst>
</response>
```

Maybe I do something wrong or maybe there is a bug?

**Comment by xavier-morera** [comment]
@Fatalityap can you capture the call that SolrNet is making to see what is the difference with the one that you are doing directly in Fiddler?

**Comment by Fatalityap** [comment]
@xavier-morera  I have already did this. There is captured call form Fiddler in my sample. But for some reason response is not correctly parsed maybe.

**Comment by xavier-morera** [comment]
Ok, thanks for clarifying. I thought that was a call done manually. Do you know how to debug SolrNet? Can you take a quick look?

**Comment by xavier-morera** [comment]
@ariasjose you worked on this if I recall. Do you see also the same behavior?

**Comment by ariasjose** [comment]
Hi,
Having spellcheck.extendedResults=true produces the frequency to be inserted in the response for each suggestion. That output isn't being parsed correctly. As a workaround set spellcheck.extendedResults=false.
I'll correct this so the fix can be included in future release.

**Comment by ariasjose** [comment]
Pull request #398 has been created.
@xavier-morera would you please check this?

**Comment by xavier-morera** [comment]
@ariasjose I will soon. @Fatalityap can you confirm as well if these changes work for you? Do you have your own fork?

**Comment by Fatalityap** [comment]
Hi @xavier-morera, hi @ariasjose  When I forced set spellcheck.extendedResults=false (add it as mandatory parameter to all queries) suggestions parsed correctly. But initially I don't set spellcheck.extendedResults at all. But thanks for workaround.

**Comment by gjunge** [comment]
Fixed by #398  . Will be part of 1.0.7
